frontend/src/types/index.ts

```diff
@@ -103,3 +98,24 @@ export interface DirectoryData {
     tooling: Tools;
   };
 }
+
+export enum DatabaseType {
+  Dynamo,
+  Disk,
+}
+
+export interface Database {
```

Comment:
Just a thought on naming, this type of pattern is often called a `Repository` pattern so `Repository` may be a more suitable name rather than `Database`. Here is a [blog post](https://medium.com/@erickwendel/generic-repository-with-typescript-and-node-js-731c10a1b98e) that explains the pattern and walks through a Typescript implementation.